planar/simplify/douglaspeucker.go

```diff
@@ -17,19 +17,22 @@ type DouglasPeucker struct {
 }
 
 func (dp DouglasPeucker) Simplify(ctx context.Context, linestring [][2]float64, isClosed bool) ([][2]float64, error) {
-	return dp.simplify(ctx, 0, linestring, isClosed)
+	ret := make([][2]float64, 0, len(linestring))
+	return dp.simplify(ctx, 0, linestring, isClosed, ret)
 }
 
-func (dp DouglasPeucker) simplify(ctx context.Context, depth uint8, linestring [][2]float64, isClosed bool) ([][2]float64, error) {
+func (dp DouglasPeucker) simplify(ctx context.Context, depth uint8, linestring [][2]float64, isClosed bool, ret [][2]float64) ([][2]float64, error) {
 
 	// helper function for debugging and tracing the code
 	var printf = func(msg string, depth uint8, params ...interface{}) {
-		if debug {
-			ps := make([]interface{}, 1, len(params)+1)
-			ps[0] = depth
-			ps = append(ps, params...)
-			logger.Printf(strings.Repeat(" ", int(depth*2))+"[%v]"+msg, ps...)
-		}
+		ps := make([]interface{}, 1, len(params)+1)
```

Comment:
I know this was originally a `var func`, but since you are not closing over any of the parameters of `simplify` would it make sense to push it up to the main scope?